packages/dx-react-scheduler-demos/src/demo-sources/scheduler-all-day-panel/material-ui/week-view.jsx

```diff
@@ -18,6 +22,30 @@ export default class Demo extends React.PureComponent {
       data: appointments,
       currentDate: new Date('2018-06-27'),
     };
+
+    this.onCommitChanges = this.commitChanges.bind(this);
+  }
+
+  commitChanges({ added, changed, deleted }) {
+    let { data } = this.state;
+    if (added) {
+      const startingAddedId = data.length > 0 ? data[data.length - 1].id + 1 : 0;
+      data = [
+        ...data,
+        {
+          id: startingAddedId,
+          ...added,
+        },
+      ];
+    }
+    if (changed) {
+      data = data.map(appointment => (
+        changed[appointment.id] ? { ...appointment, ...changed[appointment.id] } : appointment));
+    }
+    if (deleted) {
+      data = data.filter(appointment => appointment.id !== deleted);
+    }
+    this.setState({ data });
```

Comment:
Use callback version of `setState`.

https://reactjs.org/docs/react-component.html#setstate
> This  makes reading this.state right after calling setState() a potential pitfall...

Reply:
I change to the callback version in all editing demos. This demo will be removed.